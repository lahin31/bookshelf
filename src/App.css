.container {
  width: 95%;
  margin: 0 auto;
  display: flex;
}

.vertical_navbar {
  flex: 1;
  margin-top: 27px;
}

.vertical_navbar a {
  background-color: #eee;
  color: black;
  display: block;
  padding: 12px;
  text-decoration: none;
}

.vertical_navbar a:hover {
  background-color: #ccc;
}