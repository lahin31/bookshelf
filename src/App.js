/** @jsx jsx */
import { Component } from 'react';
import { Router } from "@reach/router";
import { jsx, css } from "@emotion/core";
import './App.css';
import BookContext from "./context/book-context";
import Home from "./pages/Home";
import ReadingBooks from "./pages/ReadingBooks";
import FinishedBooks from './pages/FinishedBooks';
import Navbar from './components/Navbar'

class App extends Component {
  state = {
    readingBooks: [],
    finishedReadingBooks: []
  }

  addBookToReading = book => {
    let books = [...this.state.readingBooks];
    book.readingList = true;
    books.push(book);
    this.setState({ readingBooks: books });
  }

  removeBookFromReading = (id, complete = false) => {
    let books = [...this.state.readingBooks];
    let indexOfDeletedBook = books.findIndex(book => book.id === id);
    books[indexOfDeletedBook].readingList = false;
    if(complete) {
      books[indexOfDeletedBook].complete = true;
    }
    books.splice(indexOfDeletedBook, 1);
    this.setState({
      readingBooks: books
    });
  }

  addBookToFinished = book => {
    let books = [...this.state.finishedReadingBooks];
    book.readingList = false;
    book.complete = true;
    books.push(book);
    this.setState({ finishedReadingBooks: books });
    this.removeBookFromReading(book.id, true);
  }

  render() {
    const { readingBooks, finishedReadingBooks } = this.state;
    return (
      <BookContext.Provider value={{
        readingBooks,
        finishedReadingBooks,
        addBookToReading: this.addBookToReading,
        removeBookFromReading: this.removeBookFromReading,
        addBookToFinished: this.addBookToFinished
      }}>
        <div className="container">
          <Navbar
            readingBooks={readingBooks}
            finishedReadingBooks={finishedReadingBooks}
          />
          <div className="vertical_lists" css={css`
            flex: 2;
          `}>
            <Router>
              <Home path="/" />
              <ReadingBooks path="reading_books" />
              <FinishedBooks path="finished_books" />
            </Router>
          </div>
        </div>
      </BookContext.Provider>
    );
  }
}

export default App;
