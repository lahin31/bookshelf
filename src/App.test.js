import React from 'react';
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect'
import App from "./App";

afterEach(cleanup);

// making a snapshot
it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("tests the length of the two state", () => {
  const ref = React.createRef();
  render(<App ref={ref} />);
  expect(ref.current.state.readingBooks.length).toEqual(0);
  expect(ref.current.state.finishedReadingBooks.length).toEqual(0);
});