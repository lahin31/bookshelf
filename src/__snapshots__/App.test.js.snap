// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <nav
      class="vertical_navbar"
    >
      <div
        class="css-1hgvmcu"
      >
        <a
          aria-current="page"
          href="/"
          style="color: red;"
        >
          All Available Books
        </a>
        <a
          href="/reading_books"
          style="color: blue;"
        >
          Reading Books(0)
        </a>
        <a
          href="/finished_books"
          style="color: blue;"
        >
          Finished Books(0)
        </a>
      </div>
    </nav>
    <div
      class="vertical_lists css-1scbww"
    >
      <div
        role="group"
        style="outline: none;"
        tabindex="-1"
      >
        <div
          class="all_books_wrapper css-hcdvcr"
        >
          <div
            class="title_wrapper"
          >
            <h1
              class="css-h2902l"
              data-testid="pageTitle"
            >
              All Available Books
            </h1>
          </div>
          <div
            class="all_books"
            data-testid="all_books"
          >
            <input
              class="css-13ywnoh-BookSearchField"
              placeholder="Search by Book Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
